File: ruby/lesson 4/counting_sort.rb
def counting_sort(nums, max)
    freq = Array.new(max + 1, 0)
    nums.each { |num| freq[num] += 1 }

    current = 0
    for i in 0..freq.size - 1
        for count in 0..freq[i] - 1
            nums[current] = i
            current += 1
        end
    end
end

File: java/lesson 4/CountingSort.java
public class CountingSort {
    void countingSort(int[] nums, int max) {
        int[] freq = new int[max + 1];
        for (int num : nums) {
            freq[num]++;
        }
        
        int current = 0;
        for (int i = 0; i < freq.length; i++) {
            for (int count = 0; count < freq[i]; count++) {
                nums[current] = i;
                current++;
            }
        }
    }    
}
